python/whylogs/api/logger/experimental/logger/actor/actor.py

```diff
@@ -0,0 +1,210 @@
+import logging
+import queue
+import signal
+import time
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Generic, List, Optional, Type, TypeVar, Union
+
+from whylogs.api.logger.experimental.logger.actor.list_util import type_batched_items
+from whylogs.api.logger.experimental.logger.actor.signal_util import suspended_signals
+
+QueueMessageType = TypeVar("QueueMessageType")
+
+
+class QueueWrapper(Generic[QueueMessageType]):
+    @abstractmethod
+    def send(self, message: QueueMessageType, timeout: float = 0.1) -> None:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def send_many(self, message: QueueMessageType, timeout: float = 0.1) -> None:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def get(self, timeout: float = 0.1) -> Optional[QueueMessageType]:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def get_many(self, timeout: float = 0.1, max: Optional[int] = None) -> List[QueueMessageType]:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def size(self) -> int:
+        raise NotImplementedError()
+
+
+class CloseMessage:
+    pass
+
+
+@dataclass
+class QueueConfig:
+    max_batch_size: int = 50_000
+    """
+    Maximum number of messages to process in a single batch.
```

Comment:
this should follow a [sphynx-like documentation](https://github.com/whylabs/whylogs/blob/mainline/python/whylogs/api/writer/whylabs.py#L169), as these strings won't appear on a `help(QueueConfig)` properly for example. i.e.:

```python
class QueueConfig:
    """
    This is the interface for (...) and contains (...)

    :param max_batch_size: Maximum number of messages to process in a single batch, defaults to 50_000
    :type max_batch_size: int

    :raises [ErrorType]: [ErrorDescription]

    :return: [ReturnDescription]
    :rtype: [ReturnType]
    """

    max_batch_size: int = 50_000
    (...)
```

Reply:
python...